test/functional/core/job_spec.lua

```diff
@@ -308,8 +317,15 @@ describe('jobs', function()
     nvim('command', 'unlet g:job_opts.on_exit')
     nvim('command', 'let g:job_opts.user = 5')
     nvim('command', [[call jobstart('echo "foo"', g:job_opts)]])
-    eq({'notification', 'stdout', {5, {'foo', ''}}}, next_msg())
-    eq({'notification', 'stdout', {5, {''}}}, next_msg())
+    expect_msg_seq(
+      { {'notification', 'stdout', {5, {'foo', ''} } },
+        {'notification', 'stdout', {5, {''} } }
+      },
+      -- Alternative sequence:
+      { {'notification', 'stdout', {5, {'foo'} } },
+        {'notification', 'stdout', {5, {'', ''} } }
+      }
```

Comment:
`expect_msg_seq()` accepts multiple sequences and will call `next_message()` enough times to compare the shortest sequence. If longer sequences are tried, it will call `next_message()` again as needed.